pkg/drivers/commonconfig.go

```diff
@@ -299,11 +302,19 @@ func GetNode(ctx context.Context, cr csmv1.ContainerStorageModule, operatorConfi
 
 	nodeYaml.DaemonSetApplyConfig.Spec.Template.Spec.Containers = newcontainers
 
+	updatedCr, err := SetSDCinitContainers(ctx, cr, ct)
```

Comment:
Since this is in the common path for GetNode for all platforms, have you checked whether install of say powerstore driver, powermax driver is not affected by this addition ? I think there might be some impact because SetSDCInitContainers does call GetMDMFromSecret and the secret for other platforms will have a different format. Please test.

Reply:
This issue can be addressed using the following:

```go
var updatedCr csmv1.ContainerStorageModule
if cr.Spec.Driver.CSIDriverType == "powerflex" {
    updatedCr, err = SetSDCinitContainers(ctx, cr, ct)
    if err != nil {
        log.Errorw("Failed to set SDC init container", "Error", err.Error())
        return nil, err
    }
}
```

Reply:
I have modified the code so that the SetSDCInitContainers function is only called for PowerFlex. This ensures that other drivers are not impacted, as the function will not be invoked for them.